src/codegen/generators/expression_generator.rs

```diff
@@ -1193,12 +1204,24 @@ impl<'ink, 'b> ExpressionCodeGenerator<'ink, 'b> {
                 .find_member(function_name, name)
                 .ok_or_else(|| Diagnostic::unresolved_reference(name, left.get_location()))?;
             let index = parameter.get_location_in_parent();
-            self.generate_call_struct_argument_assignment(&CallParameterAssignment {
-                assignment_statement: right,
-                function_name,
-                index,
-                parameter_struct,
-            })?;
+
+            // don't generate param assignments for empty statements, with the exception
+            // of VAR_IN_OUT params - they need an address to point to
+            let is_auto_deref = matches!(
```

Comment:
It might make sense to return an error on the empty statement case and stop codegen, although it does not really matter here. Just thinking outould